### PR TITLE
updates error message to run pytest instead of tests.py standalone

### PR DESCRIPTION
**Pull request by eseiver**
## PR Summary
updates the error message in `NoopTestCommand` to recommend using pytest instead of tests.py

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [x] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -121,7 +121,7 @@
 class NoopTestCommand(TestCommand):
     def __init__(self, dist):
         print("Matplotlib does not support running tests with "
-              "'python setup.py test'. Please run 'python tests.py'.")
+              "'python setup.py test'. Please run 'pytest'.")
```
Comment (ImportanceOfBeingErnest) [inline setup.py]: According to [the docs](https://matplotlib.org/devel/testing.html#running-the-tests) you can run the tests with `py.test` or `python tests.py`.  
Should the latter not be the case, one would need to update the docs as well. Else, the error could probably read 

> Please run 'python tests.py' or 'py.test'.

Comment (Carreau) [reply setup.py]: I believe there was an attempt to migrate to `py.test`, and to deprecate disincentive `python tests.py`, in particular the documentation does not mention how to run a single test using `python tests.py`. So it seem to me that directly giving the right invocation.

If you look at previous PRs and comment (like https://github.com/matplotlib/matplotlib/pull/8026/files#r101600628), there seem to be some consensus that `pytest` is prefered and should replace `python tests.py` where possible.

Also

```
In [1]: import this
...
There should be one-- and preferably only one --obvious way to do it.
...
```

**Comment by NelleV** [comment]
Thanks a lot for the contribution @eseiver ! As you may see, I've pushed some changes to your branch as we discussed in person earlier today. I think we've almost done with this pull request. @ivanov has agreed to review it as well.
I've left tests.py so far, but if all the functionalities are covered with pytest (which I believe they are), we should also be a go to remove it!

**Comment by tacaswell** [comment]
*grumble*  I still like `python tests.py` better, but I think all of the configuration we used to do there has moved in various configuration files and we no longer support any version of Matplotlib that uses `nose` so I am out of technical reasons.

Would it be better to do `python -mpytest`?

**Comment by choldgraf** [comment]
@tacaswell how do you feel about nailing this down in a subsequent PR? I think this PR is an improvement and tests look happy so I think we should let @eseiver off the hook for more nuanced testing changes :-)

File: setup.py (hunk 1 above)
Comment (tacaswell) [reply setup.py]: @Carreau all of the pytest CLI args pass through.

**Comment by tacaswell** [comment]
This is fine with me.

**Comment by choldgraf** [comment]
ok cool, I'm gonna merge this one! Thanks @eseiver for your first matplotlib commit :-D

File: setup.py (hunk 1 above)
Comment (Carreau) [reply setup.py]: > ll of the pytest CLI args pass through.

Oh, ok, well the old doc was not saying that, and giving innacurate information. 
Thanks for merging. I think that having pytest everywhere but still have the `python tests.py` as a alias is great !

**Pull request merged**